drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/runner/DMNScenarioRunnerHelper.java

```diff
@@ -90,6 +97,34 @@ public void verifyConditions(SimulationDescriptor simulationDescriptor,
         }
     }
 
+    protected void verifyInputConstraints(List<DMNMessage> messages, List<ScenarioGiven> givens) {
+
+        messages.stream()
+                // filter out invalid messages (only errors with sourceId)
+                .filter(message -> Message.Level.ERROR.equals(message.getLevel()) &&
+                        message.getSourceReference() != null)
+                .flatMap(this::retrieveInputNodeName)
+                .forEach(name -> applyErrorIfUsed(name, givens));
+    }
+
+    protected Stream<String> retrieveInputNodeName(DMNMessage message) {
+        Object sourceReference = message.getSourceReference();
+        if (sourceReference instanceof NamedElement &&
+                ((NamedElement) sourceReference).getName() != null) {
+            return Stream.of(((NamedElement) sourceReference).getName());
+        }
+        return Stream.empty();
+    }
+
+    protected void applyErrorIfUsed(String name, List<ScenarioGiven> givens) {
```

Comment:
@danielezonca 
Could you add two lines of javadoc to this method to explain its scope? The previous twos are pretty easy to understand by name, this one not.

Reply:
Done